src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/Workspace.cs

```diff
@@ -525,16 +526,16 @@ async Task ApplyInBatchAsync()
                     isActiveEditorContext: _activeEditorContextTracker.IsActiveEditorContext(ContextId),
                     isActiveConfiguration: IsPrimary);
 
-                Context.StartBatch();
-
                 try
                 {
+                    IAsyncDisposable disposableBatchScope = await Context.CreateBatchScopeAsync(cancellationToken);
+
+                    await using ConfiguredAsyncDisposable _ = disposableBatchScope.ConfigureAwait(false);
```

Comment:
Do we need the `ConfigureAwait(false)`? We should already be on a background thread.

Reply:
Yeah I think it can go.